### PR TITLE
Update

### PR DESCRIPTION
**Pull request by GTwhy**

### CODE REVIEW

File: src/sanity_side.rs [hunk 1]
```diff
@@ -57,12 +63,25 @@ fn main() -> anyhow::Result<()> {
...
-        RwLock::new(vec![]);
-    static ref MR_MAP: RwLock<Vec<Pin<Vec<u8>>>> = RwLock::new(vec![]);
+    static ref RDMA_MAP: Arc<RwLock<HashMap<String, Rdma>>> = Arc::new(RwLock::new(HashMap::new()));
+    static ref MR_MAP: Arc<RwLock<HashMap<usize, LocalMr>>> = Arc::new(RwLock::new(HashMap::new()));
+    static ref RUNTIME: runtime::Runtime = runtime::Runtime::new().unwrap();
+    static ref RDMA_MR_MAP: Arc<RwLock<HashMap<String, Vec<usize>>>> =
+        Arc::new(RwLock::new(HashMap::new()));
+}
+
+fn mtu_try_from_u32(value: u32) -> Result<MTU, io::Error> {
```
Comment (rogercloud) [inline src/sanity_side.rs]: usually we put try in the first place `try_mtu_from_u32`

File: src/sanity_side.rs [hunk 2]
```diff
@@ -87,284 +106,250 @@ impl Side for SideImpl {
...
+
+        rdma.ibv_connect(remote).await.unwrap();
 
         let resp = ConnectQpResponse::default();
         let f = sink.success(resp).map_err(|_| {}).map(|_| ());
         ctx.spawn(f)
     }
 
-    fn remote_read(
+    #[tokio::main]
```
Comment (rogercloud) [inline src/sanity_side.rs:227]: It'll create a runtime for each call. Is that efficient?

File: src/sanity_side.rs [hunk 3]
```diff
@@ -57,12 +63,25 @@ fn main() -> anyhow::Result<()> {
 }
 
 lazy_static! {
-    static ref DEV_MAP: RwLock<HashMap<String, rdma::ibv::IbvCtx>> = RwLock::new(HashMap::new());
-    static ref QP_MAP: RwLock<Vec<rdma::ibv::IbvQp>> = RwLock::new(vec![]);
-    static ref PD_MAP: RwLock<Vec<rdma::ibv::IbvPd>> = RwLock::new(vec![]);
-    static ref CQ_MAP: RwLock<Vec<(rdma::ibv::IbvCq, rdma::ibv::IbvEventChannel)>> =
-        RwLock::new(vec![]);
-    static ref MR_MAP: RwLock<Vec<Pin<Vec<u8>>>> = RwLock::new(vec![]);
+    static ref RDMA_MAP: Arc<RwLock<HashMap<String, Rdma>>> = Arc::new(RwLock::new(HashMap::new()));
```
Comment (rogercloud) [inline src/sanity_side.rs:66]: Is it necessary to use tokio lock?

File: src/sanity_side.rs (hunk 2 above)
Comment (GTwhy) [reply src/sanity_side.rs:227]: That is not efficient and just for the convenience of coding.
We can also use only one staic `RUNTIME` in every call, such as:
```
fn remote_read(){
  RUNTIME.block_on/spawn(
    async {...}
  )
}
```
Considering that this is a project for testing, convenience is given priority so I did this. 
What is your advise or do you have any other ideas.

File: src/sanity_side.rs (hunk 3 above)
Comment (GTwhy) [reply src/sanity_side.rs:66]: Yes, we need tokio lock in `local_recv` call, otherwise we can't spawn the task.

File: src/sanity_side.rs (hunk 2 above)
Comment (rogercloud) [reply src/sanity_side.rs:227]: Agree. Let's change it when performance is too bad.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**